src/sentry/tasks/auth.py

```diff
@@ -57,47 +59,70 @@ def email_unlink_notifications(org_id, actor_id, provider_key):
         member.send_sso_unlink_email(actor, provider)
 
 
-@instrumented_task(
-    name="sentry.tasks.remove_2fa_non_compliant_members",
-    queue="auth",
-    default_retry_delay=60 * 5,
-    max_retries=5,
-)
-def remove_2fa_non_compliant_members(org_id, actor_id=None, actor_key_id=None, ip_address=None):
-    org = Organization.objects.get(id=org_id)
-    actor = User.objects.get(id=actor_id) if actor_id else None
-    actor_key = ApiKey.objects.get(id=actor_key_id) if actor_key_id else None
+class OrganizationComplianceTask(abc.ABC):
+    """Remove members who don't comply with a new org requirement."""
 
-    for member in OrganizationMember.objects.select_related("user").filter(
-        organization=org, user__isnull=False
-    ):
-        if not Authenticator.objects.user_has_2fa(member.user):
-            _remove_2fa_non_compliant_member(
-                member, org, actor=actor, actor_key=actor_key, ip_address=ip_address
-            )
+    log_label = ""
 
+    @abc.abstractmethod
+    def is_compliant(self, member: OrganizationMember) -> bool:
+        """Check whether a member complies with the new requirement."""
+        raise NotImplementedError()
 
-def _remove_2fa_non_compliant_member(member, org, actor=None, actor_key=None, ip_address=None):
-    user = member.user
-    logging_data = {"organization_id": org.id, "user_id": user.id, "member_id": member.id}
+    @abc.abstractmethod
+    def call_to_action(self, member: OrganizationMember):
+        """Prompt a member to comply with the new requirement."""
+        raise NotImplementedError()
 
-    try:
-        member.remove_user()
-        member.save()
-    except (AssertionError, IntegrityError):
-        logger.warning("Could not remove 2FA noncompliant user from org", extra=logging_data)
-    else:
-        logger.info("2FA noncompliant user removed from org", extra=logging_data)
-        AuditLogEntry.objects.create(
-            actor=actor,
-            actor_key=actor_key,
-            ip_address=ip_address,
-            event=AuditLogEntryEvent.MEMBER_PENDING,
-            data=member.get_audit_log_data(),
-            organization=org,
-            target_object=org.id,
-            target_user=user,
-        )
+    def remove_non_compliant_members(self, org_id, actor_id, actor_key_id, ip_address):
+        org = Organization.objects.get(id=org_id)
```

Comment:
```suggestion
        org = Organization.objects.get_from_cache(id=org_id)
```
You could use `get_from_cache()` to potentially save a DB hit.

Reply:
To be clear, that would be a change to preexisting behavior on `remove_2fa_non_compliant_members`. But that should be safe.